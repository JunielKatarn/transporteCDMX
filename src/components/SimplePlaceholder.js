import React, { memo } from 'react'
import { View, Image, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'
import { Caption } from 'react-native-paper'
import { sharedStyles } from '../styles'

const SimplePlaceholder = ({ text, imageURL, fullHeight }) => {
  return (
    <View style={[sharedStyles.mx3, sharedStyles.centerChild]}>
      <View style={[styles.imageContainer, fullHeight ? styles.fullHeight : styles.halfHeight]}>
        <Image style={styles.image} source={imageURL} />
      </View>
      <Caption style={[sharedStyles.mt3, sharedStyles.textAlignCenter]}>{text}</Caption>
    </View>
  )
}

const styles = StyleSheet.create({
  imageContainer: {
    width: '100%',
  },
  image: {
    height: '100%',
    width: '100%',
    resizeMode: 'contain',
  },
  fullHeight: {
    height: 280,
  },
  halfHeight: {
    height: '58%',
  },
})

SimplePlaceholder.propTypes = {
  text: PropTypes.string,
  imageURL: PropTypes.number,
  fullHeight: PropTypes.bool,
}

export default memo(SimplePlaceholder)
