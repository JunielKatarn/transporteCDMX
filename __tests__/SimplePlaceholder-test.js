import React from 'react'
import { shallow } from 'enzyme'
import SimplePlaceholder from '../src/components/SimplePlaceholder'

const image = require('../src/assets/images/programming_code.png')

describe('SimplePlaceholder', () => {
  it('it renders correctly', () => {
    const wrapper = shallow(<SimplePlaceholder text="Placeholder" imageURL={image} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it renders correctly with fullHeight', () => {
    const wrapper = shallow(<SimplePlaceholder text="Placeholder" imageURL={image} fullHeight />)
    expect(wrapper).toMatchSnapshot()
  })
})
