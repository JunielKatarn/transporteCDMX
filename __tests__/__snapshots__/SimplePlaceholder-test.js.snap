// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePlaceholder it renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 17,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "height": "58%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/programming_code.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
  <Caption
    style={
      Array [
        Object {
          "marginTop": 17,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Placeholder
  </Caption>
</View>
`;

exports[`SimplePlaceholder it renders correctly with fullHeight 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 17,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        Object {
          "height": 280,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/programming_code.png",
        }
      }
      style={
        Object {
          "height": "100%",
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
  <Caption
    style={
      Array [
        Object {
          "marginTop": 17,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Placeholder
  </Caption>
</View>
`;
